src/services/jsTyping.ts

```diff
@@ -238,14 +223,17 @@ namespace ts.JsTyping {
                 if (!packageJson.name) {
                     continue;
                 }
-                if (packageJson.typings) {
-                    const absolutePath = getNormalizedAbsolutePath(packageJson.typings, getDirectoryPath(normalizedFileName));
+                const ownTypes = packageJson.types || packageJson.typings;
+                if (ownTypes) {
+                    const absolutePath = getNormalizedAbsolutePath(ownTypes, getDirectoryPath(normalizedFileName));
+                    if (log) log(`Package '${packageJson.name}' provides its own types.`);
                     inferredTypings.set(packageJson.name, absolutePath);
                 }
                 else {
-                    addInferredTyping(packageJson.name);
+                    packageNames.push(packageJson.name);
                 }
             }
+            addInferredTypings(packageNames, "    Found package names");

```

Comment:
I wanted to imply that this was part of the above log for "searching for typing names"; should probably also indent anything else in that part.